simplifier/operations/interface.py

```diff
@@ -57,14 +60,13 @@ def _promote_single_operand(self) -> WorldObject:
 
     def simplify(self, keep_form: bool = False) -> Optional[WorldObject]:
         """Simplify the operation by promoting single operands as well."""
-        simplified_operation = super().simplify(keep_form)
-        if simplified_operation is not None and simplified_operation != self:
-            return simplified_operation
-        operands = self.operands
-        operation = self._promote_single_operand()
-        if len(operands) == 1:
-            return operation
-        return None
+        op = super().simplify(keep_form)
+        if isinstance(op, (AssociativeOperation, CommutativeOperation)):
+            operands = op.operands
+            op._promote_single_operand()
+            if len(operands) == 1:
+                return operands[0]
```

Comment:
simplify as below

But, again, I am not sure whether we want to change the assumption that simplifies only returns a condition if the operand changes. We should think about this and change it accordingly.

Reply:
I've fixed this too.